lib/schema/schema-to-drizzle.js

```diff
@@ -0,0 +1,302 @@
+import { text, integer, real, customType } from 'drizzle-orm/sqlite-core'
+
+/**
+Convert optional properties to nullable
+@template {object} T
+@typedef {{ [K in keyof T]-?: undefined extends T[K] ? T[K] | null : T[K] }} OptionalToNull
+ */
+
+/**
+Convert a readonly array/object to writeable
+@template T
+@typedef {{ -readonly [P in keyof T]: T[P] }} Writable
```

Comment:
out of curiosity, what are the hyphens in the types?

Reply:
They are called "[mapping modifiers](https://www.typescriptlang.org/docs/handbook/2/mapped-types.html#mapping-modifiers)" and can be used to remove readonly from a type.

Reply:
oh wow, very cool!